### PR TITLE
Added support for opening applications from the pathbox

### LINKED ISSUE #5677: Support opening application from the path editbox
### Description

Similar to Windows Explorer, ability to launch a command window and execute the command typed into the navigation bar, including locating the command in the path.

### Steps To Reproduce

_No response_

### Expected behavior

Example: H:\commands is  in my PATH variable. AT that location there is a bat file m.bat. In windows explorer I can type in m in the navigation bar to open a command window which then runs H:\commands\m.bat as that is the only m.bat or m.exe in only PATH locations.

### Files Version

v1.5.0.0

### Windows Version

Version 10.0.18363.1679

### Relevant Assets

_No response_

### PR DESCRIPTION
**Pull request by StefanLobbenmeier**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #5677
- Related #7814

**Details of Changes**
Add details of changes here.
- Added LaunchApplicationFromPath, which attempts to launch the application that was entered in the path editbox.
- To be determined: if the Application was started correctly or if an error message should be shown. Right now an Error Message is shown everytime.

**Validation**
How did you test these changes?
- [ ] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Review by ThePhyscoD — DISMISSED** [review]

File: src/Files/ViewModels/NavToolbarViewModel.cs
Comment (yaichenbaum) [inline src/Files/ViewModels/NavToolbarViewModel.cs:980]: Should this go in a seperate helper?

Comment (d2dyno1) [reply src/Files/ViewModels/NavToolbarViewModel.cs:980]: @StefanLobbenmeier

Comment (StefanLobbenmeier) [reply src/Files/ViewModels/NavToolbarViewModel.cs:980]: Not sure - since the other methods also don’t use helpers it does not make much sense to do it for just this once. I guess in the long term all possible handlers of input should implement a common interface and handle it, similar to how it already done in the backend.

But if you want it in a helper now, feel free to edit. I don’t know the Codebase that well.

**Review by d2dyno1 — DISMISSED** [review]

File: src/Files/ViewModels/NavToolbarViewModel.cs
Comment (yaichenbaum) [reply src/Files/ViewModels/NavToolbarViewModel.cs:980]: We're trying to make sure new code is better organized but I'll leave this to @d2dyno1 to decide if it's a blocker.

Comment (d2dyno1) [reply src/Files/ViewModels/NavToolbarViewModel.cs:980]: I haven't touched this part yet so it's fine

**Comment by yaichenbaum** [comment]
@StefanLobbenmeier before we merge this, can you add your agreement to #7625?

**Comment by yaichenbaum** [comment]
@StefanLobbenmeier can you try this on the homepage? 
image

**Comment by gave92** [comment]
Pushed a change to address issue raised by @yair100. The error dialog should now be shown only if LaunchApplicationFromPath failed.

File: src/Files/ViewModels/NavToolbarViewModel.cs
Comment (gave92) [inline src/Files/ViewModels/NavToolbarViewModel.cs:951]: Note: I've also removed the special handling for launching terminal apps, LaunchApplicationFromPath can handle launching terminals as well.

Comment (yaichenbaum) [reply src/Files/ViewModels/NavToolbarViewModel.cs:951]: Does `wt` open in the correct path for you?

Comment (gave92) [reply src/Files/ViewModels/NavToolbarViewModel.cs:951]: It does, if it does not for you we can bring the code above back

Comment (gave92) [reply src/Files/ViewModels/NavToolbarViewModel.cs:951]: ~It does~. Noope I was trying from the toolbar, apparently we do need to pass the "Parameters" from the terminals profile.

Comment (gave92) [reply src/Files/ViewModels/NavToolbarViewModel.cs]: Restored

Comment (yaichenbaum) [reply src/Files/ViewModels/NavToolbarViewModel.cs]: Thanks!

**Review by yair100 — APPROVED** [review]
LGTM

**Comment by StefanLobbenmeier** [comment]
> Pushed a change to address issue raised by @yair100. The error dialog should now be shown only if LaunchApplicationFromPath failed.

Very good, this was the part I was struggling with since I did not know how to actually check for a success.

> @StefanLobbenmeier before we merge this, can you add your agreement to #7625?

I added it

**Comment by StefanLobbenmeier** [comment]
A bit late - but I also wanted to confirm the artifact from https://dev.azure.com/filescommunity/Files/_build/results?buildId=10171&view=artifacts&pathAsName=false&type=publishedArtifacts  (should be 7c3511d) can open IntelliJ exactly the way I wanted it to 😄

**Pull request merged**
